test/e2e/scripts/setup-instance/03-ssh.sh

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+
+printf '=%.0s' {0..79} ; echo
+set -ex
+
+test $1 || {
+    echo 'Must provide a ssh IP or DNS as $1'
+    exit 1
+}
+MACHINE=$1
+
+# get into the repo to grab the last commit hash
+test ${COMMIT_ID} || {
+    cd $(dirname $0)
+    COMMIT_ID=$(git rev-parse --verify HEAD)
+}
+
+SSH_OPTS="-o ServerAliveInterval=20 -o ConnectTimeout=6 -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -i id_rsa "
+SEND_ENV="-o SendEnv DATADOG_AGENT_IMAGE=${DATADOG_AGENT_IMAGE:-datadog/agent-dev:master}"
+
+function _ssh() {
+    ssh ${SSH_OPTS} -lcore ${MACHINE} $@
+}
+
+function _ssh_logged() {
+    ssh ${SSH_OPTS} -lcore ${MACHINE} ${SEND_ENV} /bin/bash -l -c "$@"
+}
+
+until _ssh /bin/true
+do
+    sleep 5
+done
+
+
+_ssh git clone https://github.com/DataDog/datadog-agent.git /home/core/datadog-agent || {
+    # To be able to retry
+    echo "Already cloned ?"
```

Comment:
In which case would the repo be already cloned ?

Reply:
When we are iterating on the dev target. The instance could be partially already prepared.

This practice have a lot of limitations, but accept an already cloned repo sounds reasonable to me.

Reply:
I see, I don't have much experience there but accepting an already cloned repo means potentially being out of date so I think this is legit.

Reply:
I encountered this issue while developing on this.

This mean, if you did a typo in any script, without this, you need to create a new instance to try the fix.